packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/__tests__/quanticSearchBoxInput.test.js

```diff
@@ -0,0 +1,344 @@
+        ...defaultOptions,
+        withoutSubmitButton: true,
+      });
+      await flushPromises();
+
+      const submitButton = element.shadowRoot.querySelector(
+        selectors.searchBoxSubmitBtn
+      );
+
+      expect(submitButton).toBeNull();
+    });
+  });
+
+  describe('when the placeholder property receives a custom placeholder value', () => {
+    it('should display the custom value as the searchbox placeholder', async () => {
+      const customPlaceholder = 'Custom placeholder';
+      const element = createTestComponent({
+        ...defaultOptions,
+        placeholder: customPlaceholder,
+      });
+      await flushPromises();
+
+      const input = element.shadowRoot.querySelector(selectors.searchBoxInput);
+
+      expect(input.placeholder).not.toEqual(defaultOptions.placeholder);
+      expect(input.placeholder).toEqual(customPlaceholder);
+    });
+  });
+
+  describe('when suggestions are found', () => {
+    it('should display the suggestions in the suggestions list', async () => {
+      const element = createTestComponent({
+        ...defaultOptions,
+        suggestions: mockSuggestions,
+      });
+      await flushPromises();
+
+      const suggestionsList = element.shadowRoot.querySelector(
+        selectors.searchBoxSuggestionsList
+      );
+      expect(suggestionsList).not.toBeNull();
+
+      const suggestionsListItems =
+        suggestionsList.shadowRoot.querySelectorAll('li');
+      expect(suggestionsListItems).not.toBeNull();
+
+      expect(suggestionsListItems.length).toEqual(3);
+    });
+  });
+
+  describe('when focusing on the input', () => {
+    it('should dispatch a #showsuggestions custom event', async () => {
+      const element = createTestComponent();
+      setupEventListeners(element);
+      await flushPromises();
+
+      const input = element.shadowRoot.querySelector(selectors.searchBoxInput);
+      expect(input).not.toBeNull();
+
+      await input.focus();
+
+      expect(functionsMocks.exampleHandleShowSuggestions).toHaveBeenCalledTimes(
+        1
+      );
+    });
+
+    describe('when selecting a suggestion from the suggestions list', () => {
+      it('should dispatch a #selectsuggestion event with the selected suggestion as payload', async () => {
+        const element = createTestComponent({
+          ...defaultOptions,
+          suggestions: mockSuggestions,
+        });
+        setupEventListeners(element);
+        await flushPromises();
+
+        const suggestionsList = element.shadowRoot.querySelector(
+          selectors.searchBoxSuggestionsList
+        );
+        expect(suggestionsList).not.toBeNull();
+
+        const firstSuggestion =
+          suggestionsList.shadowRoot.querySelectorAll('li')[0];
+        expect(firstSuggestion).not.toBeNull();
+
+        firstSuggestion.click();
+
+        expect(
+          functionsMocks.exampleHandleSelectSuggestion
+        ).toHaveBeenCalledTimes(1);
+
+        // @ts-ignore
+        const eventData =
```

Comment:
Why do you need a ts-ignore here?

Reply:
otherwise I get the "eventData is possibly undefined" error. The test still passes, but I get the error in the file. I tried adding a circuit breaker to make sure its not undefined : `eventData && eventData.detail.selectedSuggestion` but then I just get `Property 'detail' does not exist on type 'never'`